core/lib/Cardano/Codec/Bech32.hs

```diff
@@ -0,0 +1,32 @@
+module Cardano.Codec.Bech32 (ToBech32(..), FromBech32(..), bech32With,fromBech32With) where
+
+import Prelude
+
+import Codec.Binary.Bech32
+    ( HumanReadablePart, humanReadablePartToText )
+import Codec.Binary.Encoding
+    ( AbstractEncoding (..), encode )
+import qualified Codec.Binary.Encoding as E
+import Data.ByteString
+    ( ByteString )
+import Data.Text
+    ( Text )
+import qualified Data.Text.Encoding as T
+
+class ToBech32 a where
+  bech32 :: a -> Text
+
+class FromBech32 a where
+  fromBech32 :: Text -> Either String a
+
+bech32With :: HumanReadablePart -> ByteString -> Text
+bech32With hrp =
+    T.decodeUtf8 . encode (EBech32 hrp)
+
+fromBech32With :: HumanReadablePart -> (ByteString -> b) -> Text -> Either String b
+fromBech32With hrn con text = do
+  (hrp, bs) <- E.fromBech32 (const id) $ T.encodeUtf8 text
+  if hrp == hrn then
+    pure $ con bs
+  else
+    Left $ "Human Readable Part should be " <> show (humanReadablePartToText hrn) <> " but is " <> show (humanReadablePartToText hrp)
```

Comment:
Hi @reactormonk!

So we should probably mention that the Adrestia repositories are covered by a set of [Coding Standards](https://github.com/input-output-hk/adrestia/wiki/Coding-Standards) that we try to apply to all new code in PRs.

For the section above, there are two issues to fix:

1. The code is indented with 2 spaces. However, we use 4 space indents in our repositories (except for the `where` keyword, which is indented with 2 spaces).
2. The line length is (much) longer than our suggested line limit: 80 chars ([source](https://github.com/input-output-hk/adrestia/wiki/Coding-Standards#limit-line-length-to-80-characters)).

Note that we do sometimes allow things to exceed the 80 char limit if it would be prohibitive to adhere to it, but in many cases, it's simple to fix, so we try to adhere to it when possible.

Here's a suggested code fix:

```suggestion
fromBech32With :: HumanReadablePart -> (ByteString -> b) -> Text -> Either String b
fromBech32With hrn con text = do
    (hrp, bs) <- E.fromBech32 (const id) $ T.encodeUtf8 text
    if hrp == hrn then
        pure $ con bs
    else
        Left $ mconcat
            [ "Human Readable Part should be "
            , show (humanReadablePartToText hrn)
            , " but is "
            , show (humanReadablePartToText hrp)
            ]
```